common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -683,46 +649,14 @@ private static boolean isValidNumericChar(char c) {
      * @return true, if the string represents an IPV4 address in dotted
      *         notation, false otherwise
      */
-    public static boolean isValidIpV4Address(String value) {
-
-        int periods = 0;
-        int i;
-        int length = value.length();
-
-        if (length > 15) {
-            return false;
-        }
-        char c;
-        StringBuilder word = new StringBuilder();
-        for (i = 0; i < length; i ++) {
-            c = value.charAt(i);
-            if (c == '.') {
-                periods ++;
-                if (periods > 3) {
-                    return false;
-                }
-                if (word.length() == 0) {
-                    return false;
-                }
-                if (Integer.parseInt(word.toString()) > 255) {
-                    return false;
-                }
-                word.delete(0, word.length());
-            } else if (!Character.isDigit(c)) {
-                return false;
-            } else {
-                if (word.length() > 2) {
-                    return false;
-                }
-                word.append(c);
-            }
-        }
-
-        if (word.length() == 0 || Integer.parseInt(word.toString()) > 255) {
-            return false;
-        }
-
-        return periods == 3;
+    @SuppressWarnings("DuplicateBooleanBranch")
+    public static boolean isValidIpV4Address(String ip) {
+        int len = ip.length(), i, j;
```

Comment:
please put the declaration of `i` and `j`on separate lines to make this more clear.